lib/mongo_batch.rb

```diff
@@ -15,16 +15,20 @@ def batches
         processed_so_far = offset
 
         offset.step(by: batch_size, to: to - batch_size).each do |offset|
-          yielder << query.order_by(order_by).limit(batch_size).skip(offset)
+          yielder << with_order.limit(batch_size).skip(offset)
           processed_so_far += batch_size
         end
 
         if processed_so_far < to
           last_limit = to - processed_so_far
-          yielder << query.order(order_by).limit(last_limit).skip(processed_so_far)
+          yielder << with_order.limit(last_limit).skip(processed_so_far)
         end
       end
     end
+
+    def with_order
+      query.all.options.sort ? query.all : query.order_by(order_by)
```

Comment:
After discussion offline, decided to retain this requirement for the moment – it allows us to pass a class which includes `Mongoid::Document` as well as an actual scope.